validator/project_validator.go

```diff
@@ -580,11 +585,22 @@ func ensureReferentialIntegrity(project *model.Project, distroIDs []string, dist
 								"project '%s' references a non-existent "+
 								"task '%s'", buildVariant.Name,
 								project.Identifier, task.Name),
+							Level: Error,
 						},
 					)
 				}
 			}
 			buildVariantTasks[task.Name] = true
+
+			if _, ok := taskGroupTaskSet[task.Name]; ok {
+				errs = append(errs,
+					ValidationError{
+						Message: fmt.Sprintf("task '%s' in build variant '%s' is already referenced in a task group",
+							task.Name, buildVariant.Name),
+						Level: Error,
```

Comment:
If current projects already rely on this behavior, we should change this to Warning level